filters/auth/tokeninfo_test.go

```diff
@@ -245,21 +230,58 @@ func TestOAuth2Tokeninfo(t *testing.T) {
 				return
 			}
 
-			if ti.hasAuth {
+			if ti.auth != "" {
 				req.Header.Set(authHeaderName, authHeaderPrefix+ti.auth)
 			}
 
-			rsp, err := http.DefaultClient.Do(req)
-			if err != nil {
-				t.Error(err)
+			for i := 0; i < N; i++ {
+				rsp, err := http.DefaultClient.Do(req)
+				if err != nil {
+					t.Fatal(err)
+				}
+				rsp.Body.Close()
+
+				if rsp.StatusCode != ti.expected {
+					t.Errorf("auth filter failed got=%d, expected=%d, route=%s", rsp.StatusCode, ti.expected, r)
+				}
 			}
 
-			defer rsp.Body.Close()
+			if ti.expectRequests != authRequests {
+				t.Errorf("expected %d auth requests, got: %d", ti.expectRequests, authRequests)
+			}
+		})
+	}
+}
+
+func TestOAuth2TokeninfoInvalidArguments(t *testing.T) {
+	for _, ti := range []struct {
+		msg      string
+		authType string
+		args     []interface{}
+	}{
+		{
+			msg:      "missing arguments",
+			authType: filters.OAuthTokeninfoAnyScopeName,
+		}, {
+			msg:      "anyKV(): wrong number of arguments",
+			authType: filters.OAuthTokeninfoAnyKVName,
+			args:     []interface{}{"not-matching-scope"},
+		}, {
+			msg:      "allKV(): wrong number of arguments",
+			authType: filters.OAuthTokeninfoAllKVName,
+			args:     []interface{}{"not-matching-scope"},
+		},
+	} {
+		t.Run(ti.msg, func(t *testing.T) {
+			tio := TokeninfoOptions{
+				URL:     "https://authserver.test/info",
```

Comment:
in general I think we should use `.local` as DNS suffix, for example `https://authserver.test.local/info` to omit that one day `.test` is a valid public DNS domain, but I think it's also fine like this in this PR.

Reply:
`.test` is a reserved TLD https://www.rfc-editor.org/rfc/rfc2606.html